src/programs/match_template/match_template.cpp

```diff
@@ -1351,22 +1221,25 @@ bool MatchTemplateApp::DoCalculation( ) {
         best_defocus.Resize(original_input_image_x, original_input_image_y, 1, central_average);
 
         // Make sure there is enough space allocated for all results
-        number_of_result_floats += max_intensity_projection.real_memory_allocated * number_of_output_images;
+        number_of_result_floats += max_intensity_projection.real_memory_allocated * cistem::match_template::number_of_output_images;
         number_of_result_floats += histogram_number_of_points * sizeof(long) / sizeof(float); // histogram are longs
 
         float* result = new float[number_of_result_floats];
         // Not zero floating this array since all additions are assignments. This can help to expose any indexing errors.
 
-        result[0] = max_intensity_projection.logical_x_dimension;
-        result[1] = max_intensity_projection.logical_y_dimension;
-        result[2] = max_intensity_projection.real_memory_allocated;
-        result[3] = histogram_number_of_points;
-        result[4] = actual_number_of_ccs_calculated;
-        result[5] = (float)sqrt_input_pixels;
-        //        result[5] = original_input_image_x;
-        //        result[6] = original_input_image_y;
+        {
+            using cm_t = cistem::match_template::Enum;
+
+            result[cm_t::image_size_x]                = max_intensity_projection.logical_x_dimension;
+            result[cm_t::image_size_y]                = max_intensity_projection.logical_y_dimension;
+            result[cm_t::image_real_memory_allocated] = max_intensity_projection.real_memory_allocated;
+            result[cm_t::number_of_histogram_bins]    = histogram_number_of_points;
+            result[cm_t::number_of_cccs]              = actual_number_of_ccs_calculated;
+            result[cm_t::ccc_scalar]                  = (float)sqrt_input_pixels;
+            result[cm_t::pixel_size]                  = pixel_size;
```

Comment:
Ok, this is now a change to the communication between Worker and Master, but should not affect the GUI , right?

And I just checked and the answer is yes. Workers send ProgramDefinedResults to the Master and the Master sends TemplateMatchingResults to the GUI.

Reply:
Right, afaik that is right